libs/execution_context/ebpf_maps.c

```diff
@@ -1212,6 +1369,87 @@ _delete_lru_hash_map(_In_ _Post_invalid_ ebpf_core_map_t* map)
 static ebpf_result_t
 _delete_hash_map_entry(_Inout_ ebpf_core_map_t* map, _In_ const uint8_t* key);
 
+/**
+ * @brief Walk the cold lists, removing secondary keys and finding the oldest primary key.
+ *
+ * @param[in] lru_map
+ * @return Either the oldest primary key or NULL if no primary keys are present.
+ */
+static ebpf_lru_entry_t*
+_reap_lru_cold_lists(ebpf_core_lru_map_t* lru_map)
+{
+    ebpf_lru_entry_t* oldest_entry = NULL;
+    uint64_t oldest_timestamp = UINT64_MAX;
+    bool non_empty_partition_found = true;
+
+    // Scan all partitions until we find a non-empty cold list or we have an oldest entry.
+    while (!oldest_entry && non_empty_partition_found) {
```

Comment:
Is this loop guaranteed to terminate? If this routine is entered (i.e. the hash table _was_ full) and then some other thread deletes every entry from the map, won't this loop scan forever? Intuitively, it will not find an oldest entry nor will it find a non-empty partition, and from reading the code, I don't see a limit to the loop iterations on an empty LRU.